### ISSUE TITLE
KbNet SIDD model, random color artifacts, mostly on white and dark areas

### ISSUE THREAD

**Issue by zelenooki87** [chaiNNer-org/chaiNNer#2531]
![image](https://github.com/chaiNNer-org/chaiNNer/assets/75204635/73cf4527-9e24-4e0a-a271-52026c9dc6b1)
I have a problem with kbnet architecture, latest spandrel and chainner nightly version.
no matter if I choosee fp16 or full precision mode for pytorch....
Running chainner on RTX 3090, 32 GB RAM etc.
Please advice.

**Comment by RunDevelopment** [comment]
This looks like an overflow issue, but I wonder how this is possible. Spandrel recently added clipping to all models, so this should be impossible...

Can you send us the image + name of the model, so we can reproduce this?

**Comment by zelenooki87** [comment]
Yes, I used sidd.pth model of KBNet for denoising, example of artifacts and input tiff file
[https://mega.nz/folder/spxzlCpS#cVKAOaFtSmGWwY1bEJ54yQ](url)
Please fix this if you can. If it is not on the architecture model side.
Thanks

**Comment by zelenooki87** [comment]
@RunDevelopment Do you experience similar on your end? The SenseNoise model of KBNet works as intended, but I get better results with the sidd model. It would be absolutely phenomenal if you could fix this "little thing" if it's within your code. I've found that SCUNET real gan in combination with sidd, a 50/50% blend mode gives phenomenal results when paired with topaz gigapixel ai for resolution enhancement. For the vast majority of photos in my gallery. Thank you very much in advance.

**Comment by RunDevelopment** [comment]
I reproduced it. I also played around with it a little, and it's a problem with the model. The artifacts are *highly* sensitive to the input. E.g. if you move shift the entire image over by 1px, you'll see very different artifacts. 

I also ruled at overflow because of a problem with spandrel or chainner. The really does give us back these artifacts as is.

**Comment by zelenooki87** [comment]
Can you add support for the KBNet blind denoising (color gaussian noise 50) model? I would also like the original KBNet models (https://github.com/zhangyi-3/KBNet/blob/main/Denoising/pretrained_models/README.md) to be supported in Chainer as well. Thank you.

**Comment by RunDevelopment** [comment]
Do they not work? Our implementation should theoretically just work with them. Is that not the case?

**Comment by zelenooki87** [comment]
Unfortunately, even NAFNet-SIDD-width64 produces artifacts, albeit on a smaller number of images when I run it in batch mode. I wanted to try Baseline-SIDD-width64 from NafNet too, but it's not supported. As for the mentioned models, the KBNet repository itself lists OneDrive links for downloading individual models, each around 500MB. Out of KBNet's denoise models, only SenseNoise works for me without artifacts. The Gaussian color models throw out huge patch artifacts, specifically I tried gau_color_50.pth in both fp16 and fp32 modes (see attachment). When I wrote the initial models, I referred to the readme file from the KBNet repository which leads to Google Drive models. They are significantly smaller in size. I assume they were prototypes, but I wanted to test them anyway. Thanks. (color gaussian 50 (KBnet) https://i.postimg.cc/pdW6wzL2/Snimak-ekrana-2024-01-29-191008.png)

**Comment by RunDevelopment** [comment]
It is so weird that all these model *kind of* work. Even the image for color gaussian 50 (KBnet) you posted *kind of* works in some regions. This is so weird because I would expect the image to be complete garbage if I implemented KBNet incorrectly. I don't get it...

I guess I'll set up the original KBNet repo tomorrow and tests these images with the original architecture code.

> I wanted to try Baseline-SIDD-width64 from NafNet too, but it's not supported.

Yes, baseline is a different architecture (technically). I thought it wouldn't be useful, so I didn't implement it. Do you think we should support it too?

**Comment by zelenooki87** [comment]
There's no need for baseline models. I just tried the nafnet repository in a Conda virtual environment. Without the tiling function, nafnet doesn't produce artifacts. In Chainer nafnet, when I set it to no-tiling, it also doesn't create artifacts on the slices where it used to (it was causing problems on auto-tiling mode before). KBNet is much more demanding for the no-tiling function in terms of VRAM usage, so I couldn't get any further with it. If you need an example of image where nafnet was producing artifacts, let me know, I'll send it to you.
